env0/resource_environment.go

```diff
@@ -563,6 +563,11 @@ func assertDeploymentTriggers(autoDeployByCustomGlob string, continuousDeploymen
 			return diag.Errorf("cannot set auto_deploy_by_custom_glob when auto_deploy_on_path_changes_only is disabled")
 		}
 	}
+
+	if autoDeployOnPathChangesOnly && !continuousDeployment && !pullRequestPlanDeployments {
```

Comment:
I hope this use-case is enough. Please let me know if there are additional use-cases required.

Reply:
Thinking about this again, this client doesn't really have enough information to perform this validation. For example, in BE we might have a policy for an organisation that states what the default `continuousDeployment` and `pullRequestPlanDeployments` for new environments is.

I think the issue here is just that the backend validation doesn't exist on `create`, and does exist on `update`.

So we don't need this validation at all IMO.
We still need to remove the `true` default, and only send `autoDeployOnPathChangesOnly` on `update` if it has changed, as you've done in this PR